replications/internal/queue_management.go

```diff
@@ -133,20 +133,10 @@ func (rq *replicationQueue) run() {
 		case <-rq.done: // end the goroutine when done is messaged
 			return
 		case <-rq.receive: // run the scanner on data append
-			for {
-				_, err := rq.SendWrite(rq.writeFunc)
-
-				if err != nil {
-					if err == io.EOF {
-						// No more data
-						// Handle this gracefully, as it is an expected error to receive
-					} else {
-						// Crash out the server if an unhandleable error occurs in SendWrite()
-						// Should ideally never be hit
-						panic(1)
-					}
-					break
-				}
+			if rq.SendWrite(rq.writeFunc) {
+				// Crash out the server if an unhandleable error occurs in SendWrite()
+				// Should ideally never be hit
+				panic(1)
```

Comment:
Hrm, miscommunication... the intent is to never need to `panic` here because all errors are handled. The `bool` output of `SendWrite` signals whether or not there might be more data to read, so we can write something like:
```
for rq.SendWrite(rq.writeFunc) {}
```
to drain out all data

Reply:
(For context, I'm thinking to mimic the semantics of common methods/patterns like `Next()` by returning the `bool`)